vulnerabilities/importers/istio.py

```diff
@@ -0,0 +1,205 @@
+            processed_data = self.process_file(f)
+            if processed_data:
+                advisories.extend(processed_data)
+        return self.batch_advisories(advisories)
+
+    def get_versions_for_pkg_from_range_list(self, version_range_list):
+        """Takes a list of version ranges(affected) of a package
+        as parameter and returns a tuple of safe package versions and
+        vulnerable package versions"""
+
+        safe_pkg_versions = []
+        vuln_pkg_versions = []
+        all_version = self.version_api.get("istio/istio")
+        if not version_range_list:
+            return all_version, []
+        version_ranges = {RangeSpecifier(r) for r in version_range_list}
+        for version in all_version:
+            if any([version in v for v in version_ranges]):
+                vuln_pkg_versions.append(version)
+
+        safe_pkg_versions = set(all_version) - set(vuln_pkg_versions)
+        return safe_pkg_versions, vuln_pkg_versions
+
+    def get_data_from_yaml_lines(self, yaml_lines):
+        """Return a mapping of data from a iterable of yaml_lines
+        for example :
+            ['title: ISTIO-SECURITY-2019-001',
+            'description: Incorrect access control.','cves: [CVE-2019-12243]']
+
+            would give {'title':'ISTIO-SECURITY-2019-001',
+            'description': 'Incorrect access control.',
+            'cves': '[CVE-2019-12243]'}
+        """
+
+        return yaml.safe_load("\n".join(yaml_lines))
+
+    def get_yaml_lines(self, lines):
+        """The istio advisory file contains lines similar to yaml format .
+        This function extracts those lines and return an iterable of lines
+
+        for example :
+            lines =
+            ---
+            title: ISTIO-SECURITY-2019-001
+            description: Incorrect access control.
+            cves: [CVE-2019-12243]
+            ---
+
+        get_yaml_lines(lines) would return
+        ['title: ISTIO-SECURITY-2019-001','description: Incorrect access control.'
+        ,'cves: [CVE-2019-12243]']
+        """
+
+        for line in lines:
+            line = line.strip()
+            if line.startswith("---"):
+                continue
+            elif line.endswith("---"):
+                break
+            else:
+                yield line
+
+    def process_file(self, path):
+
+        advisories = []
+
+        data = self.get_data_from_md(path)
+
+        releases = []
+        if data.get("releases"):
+            for release in data["releases"]:
+                release = release.strip()
+                release = release.split(" ")
+                if len(release) > 2:
+                    lbound = ">=" + release[0]
+                    ubound = "<=" + release[2]
+                    releases.append(lbound + "," + ubound)
+
+        data["releases"] = releases
+
+        if not data.get("cves"):
+            data["cves"] = [""]
+
+        for cve_id in data["cves"]:
+
+            if not cve_id.startswith("CVE"):
```

Comment:
I wonder what when this condition gets satisfied ?  If it doesn't have a `CVE` then that makes it even more special and we should collect it :)

Reply:
Actually I added that condition because some files had cves with 'N/A' . I think I got it wrong we should have "" as cve for such cases am I right?

Reply:
yes we should have empty string as cve. But in any case, we don't give up the data, create cve-less advisory if required.